internal/http/server/server_handler.go

```diff
@@ -0,0 +1,138 @@
+		idx := slices.IndexFunc(urlPart, func(c string) bool { return c == "value" })
+		if len(urlPart)-idx < 3 {
+			http.Error(w, "URL is too short", http.StatusNotFound)
+			return
+		}
+		valueType := chi.URLParam(r, "type")
+		name := chi.URLParam(r, "name")
+		var body string
+		if valueType == "gauge" {
+			val, ok := storage.Gauge[name]
+			if ok {
+				body = strconv.FormatFloat(val, 'f', -1, 64)
+			} else {
+				w.WriteHeader(http.StatusNotFound)
+			}
+		} else if valueType == "counter" {
+			val, ok := storage.Counter[name]
+			if ok {
+				body = strconv.FormatInt(val, 10)
+			} else {
+				w.WriteHeader(http.StatusNotFound)
+			}
+		} else {
+			http.Error(w, "Unknown metric type", http.StatusBadRequest)
+			return
+		}
+		w.Header().Set("Content-Type", "text/plain")
+		w.Header().Set("Content-Length", strconv.Itoa(len(body)))
+		w.WriteHeader(http.StatusOK)
+		// пишем тело ответа
+		_, err := w.Write([]byte(body))
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func UpdateMetric(storage *storage.MemStorage) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			// разрешаем только POST-запросы
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		urlPart := strings.Split(r.URL.Path, "/")
+		idx := slices.IndexFunc(urlPart, func(c string) bool { return c == "update" })
+		if len(urlPart)-idx < 4 {
+			http.Error(w, "URL is too short", http.StatusNotFound)
+			return
+		}
+		valueType := chi.URLParam(r, "type")
+		name := chi.URLParam(r, "name")
+		valueStr := chi.URLParam(r, "value")
+		if valueType == "gauge" {
+			value, err := strconv.ParseFloat(valueStr, 64)
+			if err == nil {
+				storage.Gauge[name] = value
+			} else {
+				http.Error(w, "Wrong data type, float64 is expected", http.StatusBadRequest)
+				return
+			}
+		} else if valueType == "counter" {
+			value, err := strconv.ParseInt(valueStr, 10, 64)
+			if err == nil {
+				storage.Counter[name] += value
+			} else {
+				http.Error(w, "Wrong data type, int64 is expected", http.StatusBadRequest)
+				return
+			}
+		} else {
+			http.Error(w, "Unknown metric type", http.StatusBadRequest)
+			return
+		}
+		// устанавливаем заголовок Content-Type
+		// для передачи клиенту информации, кодированной в JSON
+		w.Header().Set("Content-Type", "text/plain")
+		// устанавливаем код 200
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func MainPage(storage *storage.MemStorage) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet {
+			urlPart := strings.Split(r.URL.Path, "/")
+			if len(urlPart) > 2 {
+				http.Error(w, "Wrong URL", http.StatusBadRequest)
+				return
+			}
+			if len(urlPart[1]) != 0 {
+				http.Error(w, "Wrong URL", http.StatusBadRequest)
+				return
+			}
+			wd, err := os.Getwd()
+			if err != nil {
+				panic(err)
+			}
+			tmplFile := filepath.Join(filepath.Dir(wd), "../internal/http/server/main_page_html.tmpl")
```

Comment:
CHANGE REQUEST: this is a magic string

Reply:
I've added it to the server's config and also add addititional command ine argument. Is that what you want?